CMakeLists.txt

```diff
@@ -45,12 +60,99 @@ include_directories(${LUA_INCLUDE_DIR})
-target_link_libraries(NovelRT glad::glad SDL2::SDL2main SDL2::SDL2-static SDL2::SDL2_image Freetype::Freetype glm ${ADDITIONAL_LINK_LIBRARIES})
+set(NOVELRT_HEADERS
+  src/GeoBounds.h
+  src/GeoVector.h
+  src/GraphicsCharacterRenderData.h
+  src/KeyCode.h
+  src/NovelBasicFillRect.h
+  src/NovelBasicInteractionRect.h
+  src/NovelCommonArgs.h
+  src/NovelDebugService.h
+  src/NovelImageRect.h
+  src/NovelInteractionObject.h
+  src/NovelInteractionService.h
+  src/NovelLayeringService.h
+  src/NovelObject.h
+  src/NovelObjectSortComparison.h
+  src/NovelRenderingService.h
+  src/NovelRenderObject.h
+  src/NovelRTMacroUtilities.h
+  src/NovelRunner.h
+  src/NovelStepTimer.h
+  src/NovelTextRect.h
+  src/RGBAConfig.h
+  src/wrapper.hpp
+)
 
-add_library(NovelRTLib SHARED src/NovelInteractionService.h src/NovelLayeringService.cpp src/NovelObject.cpp src/NovelObject.h src/GeoVector.h src/NovelBasicFillRect.cpp src/NovelBasicFillRect.h src/RGBAConfig.cpp src/RGBAConfig.h src/main.cpp src/NovelImageRect.cpp src/NovelImageRect.h src/NovelRunner.cpp src/NovelRunner.h src/NovelRenderingService.cpp src/NovelRenderingService.h src/NovelRunner_C.cpp src/NovelRunner_C.h src/NovelCommonArgs.h src/NovelInteractionService.cpp src/NovelRenderObject.cpp src/NovelRenderObject.h src/NovelTextRect.cpp src/NovelTextRect.h src/NovelInteractionObject.cpp src/NovelInteractionObject.h src/KeyCode.h src/NovelObjectSortComparison.h src/NovelBasicInteractionRect.cpp src/NovelBasicInteractionRect.h src/GeoBounds.cpp src/GeoBounds.h src/GraphicsCharacterRenderData.h src/NovelRTMacroUtilities.h src/NovelStepTimer.h src/NovelStepTimer.cpp src/NovelDebugService.h src/NovelDebugService.cpp)
-target_link_libraries(NovelRTLib glad::glad SDL2::SDL2main SDL2::SDL2-static SDL2::SDL2_image Freetype::Freetype glm ${ADDITIONAL_LINK_LIBRARIES})
+set(NOVELRT_SOURCES
+  src/GeoBounds.cpp
+  src/NovelBasicFillRect.cpp
+  src/NovelBasicInteractionRect.cpp
+  src/NovelDebugService.cpp
+  src/NovelImageRect.cpp
+  src/NovelInteractionObject.cpp
+  src/NovelInteractionService.cpp
+  src/NovelLayeringService.cpp
+  src/NovelObject.cpp
+  src/NovelRenderingService.cpp
+  src/NovelRenderObject.cpp
+  src/NovelRunner.cpp
+  src/NovelStepTimer.cpp
+  src/NovelTextRect.cpp
+  src/RGBAConfig.cpp
+)
+
+set(NOVELRT_LINK_LIBRARIES
+  Freetype::Freetype
+  glad::glad
+  glm
+  SDL2::SDL2-static
+  SDL2::SDL2_image
+  SDL2::SDL2main
+)
+
+if(NOT WIN32)
+    set(NOVELRT_LINK_LIBRARIES
+      ${NOVELRT_LINK_LIBRARIES}
+      m
+    )
+endif()
+
+set(NOVELRT_EXECUTABLE_HEADERS
+  ${NOVELRT_HEADERS}
+)
+
+set(NOVELRT_EXECUTABLE_SOURCES
```

Comment:
Was thinking about this.... It might be better if the exe, rather than pulling in the sources directly, instead depends on the lib (whether static or shared; although only a shared is being produced right now).

Reply:
Ticket #48 now exists to discuss this further. :)